### PR TITLE
Added new HTN operations and preconditions

### PR DESCRIPTION
**Pull request by Tornado-Technology**
## About the PR
At the code level (These operators are not used in prototypes), new prototypes have been added for various mathematical operations with float and bool, as well as playing music.

## Why / Balance
When designing complex monster behavior, I was faced with the need to create various state falsies, creating random delays that would act as an ability cooldown rather than a wait for the monster. Various distance checks, as well as sound effects.

## Technical details
Just new HTN operators, and moving some old ones to the new namespace "Math"

## Media
https://github.com/space-wizards/space-station-14/assets/54727692/0a6570ad-59f2-4a20-9692-7214980c9047

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
Move `SetFloatOperator` from `Content.Server.NPC.HTN.PrimitiveTasks.Operators` -> `Content.Server.NPC.HTN.PrimitiveTasks.Operators.Math`

### CODE REVIEW

**Comment by Tornado-Technology** [comment]
Example to use:

```yml
- type: htnCompound
  id: TestCompound
  branches:
    - tasks:
      - !type:HTNPrimitiveTask
        operator: !type:UnPullOperator
        preconditions:
          - !type:PulledPrecondition
            isPulled: true
    
   - tasks:
      - !type:HTNPrimitiveTask
        preconditions:
          - !type:KeyBoolEqualsPrecondition
            key: AbilityIsReady
            value: true
        operator: !type:PlaySoundOperator
          sound: # some example sound path

   - tasks:
      - !type:HTNCompoundTask
        task: MeleeAttackTargetCompound
    
    - tasks:
      - !type:HTNPrimitiveTask
        preconditions:
          - !type:KeyBoolEqualsPrecondition
            key: AbilityIsReady
            value: false
        operator: !type:SetBoolOperator
          targetKey: AbilityIsReady
          value: true

      - !type:HTNCompoundTask
        task: IdleCompound
```

**Review by TheShuEd — APPROVED** [review]

File: Resources/Prototypes/Entities/Mobs/Debugging/debug_counter.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- type: entity
+  parent: MobHuman
+  id: MobDebugRandomCounter
+  name: debug random counter
+  description: He can randomize
+  suffix: AI, DEBUG
+  components:
+    - type: HTN
+      rootTask:
+        task: DebugRandomCounterCompound
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Mobs/Debugging/debug_counter.yml:28]: this compounds not exist

File: Content.Server/NPC/HTN/PrimitiveTasks/Operators/PlaySoundOperator.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    [DataField(required: true)]
+    public SoundSpecifier? Sound;
+
+    public override void Initialize(IEntitySystemManager sysManager)
+    {
+        base.Initialize(sysManager);
+
+        _audio = IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<AudioSystem>();
+    }
```
Comment (metalgearsloth) [inline Content.Server/NPC/HTN/PrimitiveTasks/Operators/PlaySoundOperator.cs:18]: It's right there mate.

Comment (Tornado-Technology) [reply Content.Server/NPC/HTN/PrimitiveTasks/Operators/PlaySoundOperator.cs:18]: Oh, for example I use other HTN opertator (Say), In the future I may also do HTN, I’ll have to fix that

**Pull request merged**